nixos/modules/services/networking/pihole-ftl-setup-script.nix

```
@@ -0,0 +1,82 @@
+{
+  cfg,
+  config,
+  lib,
+  pkgs,
+}:
+
+let
+  pihole = pkgs.pihole;
+  makePayload =
+    list:
+    builtins.toJSON {
+      inherit (list) type enabled;
+      address = list.url;
+      comment = list.description;
+    };
+  payloads = map makePayload cfg.lists;
+in
+''
+  # Can't use -u (unset) because api.sh uses API_URL before it is set
+  set -eo pipefail
+  pihole="${lib.getExe pihole}"
+  jq="${lib.getExe pkgs.jq}"
+
+  # If the database doesn't exist, it needs to be created with gravity.sh
+  if [ ! -f '${cfg.stateDirectory}'/gravity.db ]; then
+    $pihole -g
+    # Send SIGRTMIN to FTL, which makes it reload the database, opening the newly created one
+    ${pkgs.procps}/bin/kill -s SIGRTMIN $(systemctl show --property MainPID --value ${config.systemd.services.pihole-ftl.name})
+  fi
+
+  source ${pihole}/usr/share/pihole/advanced/Scripts/api.sh
```

```suggestion
  source ${pihole}/share/pihole/advanced/Scripts/api.sh
```
and fix the package accordingly